### PR TITLE
SliceManipulator and ImageMapper improvements

### PR DESCRIPTION
**Pull request by agirault**
- add slicing manipulator: vtkSliceManipulator
- improve ImageMapper to:
  - clamp slicing to the volume bounds
  - allow slicing from X, Y and Z axis from the camera

### CODE REVIEW

**Comment by jourdain** [comment]
Already pushed and merged your fix regarding the `lights geometry to follow camera`

File: Sources/Interaction/Manipulators/SliceManipulator/index.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    distance += delta;
+
+    // clamp the distance to the clipping range
+    if (distance < range[0]) {
+      distance = range[0];
+    }
+    if (distance > range[1]) {
+      distance = range[1];
+    }
+    camera.setDistance(distance);
```
Comment (agirault) [inline Sources/Interaction/Manipulators/SliceManipulator/index.js]: @jourdain @floryst @martinken what are your thoughts on this?

This is the same as seen in [`vtkInteractorStyleImage`](https://github.com/Kitware/vtk-js/blob/b807950244e1ed7db1ae6858c6829ce33bc32786/Sources/Interaction/Style/InteractorStyleImage/index.js#L261-L268): this ends up calling [`vtkImageMapper.setZSliceFromCamera`](https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/ImageMapper/index.js#L19-L25) where the focal point (defined here by `distance`) is converted into a slice index to display.

The issue I see is that this distance is clamped by the clipping range of the camera, which I assume should match the bounds of the volume. However, when testing in the `imageMapper` example ([here](https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/ImageMapper/example/index.js#L36-L38), either with that interactorStyle or with the slice manipulator), I see that the z slice index goes between [-18; 216], even though the bounds of that image are [0; 200].

Thoughts? Also, does that only work for the z axis? We could reclamp the camera focalpoint by checking if the index is out of bounds in `setZSliceFromCamera`, but that doesn't seem clean to me.

**Comment by agirault** [comment]
> Already pushed and merged a fix regarding the lights geometry to follow camera

Beat me to it! Rebasing my branch.

**Comment by jourdain** [comment]
Like your changes to support any axis.

File: Sources/Interaction/Manipulators/SliceManipulator/index.js (hunk 1 above)
Comment (agirault) [reply Sources/Interaction/Manipulators/SliceManipulator/index.js]: After discussion with @jourdain , we will leave this as is, and implement a `rangeManipulator` that will be generic enough to control scalar values (like slice, window, level), from horizontal drag, vertical drag, or pinch.

**Pull request merged**